diff --git a/init/Makefile b/init/Makefile
index 8316c23be..0063e10eb 100644
--- a/init/Makefile
+++ b/init/Makefile
@@ -31,2 +31,4 @@ preempt-flag-$(CONFIG_PREEMPT_RT)	:= PREEMPT_RT
-build-version = $(or $(KBUILD_BUILD_VERSION), $(build-version-auto))
-build-timestamp = $(or $(KBUILD_BUILD_TIMESTAMP), $(build-timestamp-auto))
+#build-version = $(or $(KBUILD_BUILD_VERSION), $(build-version-auto))
+#build-timestamp = $(or $(KBUILD_BUILD_TIMESTAMP), $(build-timestamp-auto))
+build-version = 1
+build-timestamp = "Fri Jan 9 16:33:46 UTC 2026"
@@ -55 +57,2 @@ CFLAGS_version.o := -include $(obj)/utsversion-tmp.h
-include/generated/utsversion.h: build-version-auto = $(shell $(srctree)/$(src)/build-version)
+#include/generated/utsversion.h: build-version-auto = $(shell $(srctree)/$(src)/build-version)
+include/generated/utsversion.h: build-version-auto = 1
diff --git a/scripts/mkcompile_h b/scripts/mkcompile_h
index 2596f78e5..dcc5e4bca 100755
--- a/scripts/mkcompile_h
+++ b/scripts/mkcompile_h
@@ -24,3 +24,3 @@ cat <<EOF
-#define LINUX_COMPILE_BY	"${LINUX_COMPILE_BY}"
-#define LINUX_COMPILE_HOST	"${LINUX_COMPILE_HOST}"
-#define LINUX_COMPILER		"${CC_VERSION}, ${LD_VERSION}"
+#define LINUX_COMPILE_BY	"build-user"
+#define LINUX_COMPILE_HOST	"build-host"
+#define LINUX_COMPILER		"Android (10087095, +pgo, +bolt, +lto, -mlgo, based on r487747c) clang version 17.0.2 (https://android.googlesource.com/toolchain/llvm-project d9f89f4d16663d5012e5c09495f3b30ece3d2362), LLD 17.0.2"
diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 1b733ae4c..1db2d4e47 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -173 +173,2 @@ fi
-echo "$res"
+#echo "$res"
+echo "-android14-11-gfa1d6308d1fe-ab14691759"
